force-app/main/default/classes/ContactTriggerHandler.cls

```diff
@@ -0,0 +1,37 @@
+public class ContactTriggerHandler {
+
+    public static void beforeInsert(List<Contact> newContacts) {
+    // When a contact is inserted
+	// if DummyJSON_Id__c is null, generate a random number between 0 and 100 and set this as the contact's DummyJSON_Id__c value
+    List<Contact> contactsToCreate = new List<Contact>();
+        for (Contact currContact : newContacts) {
+            if (currContact.DummyJSON_Id__c == null) {
+            currContact.DummyJSON_Id__c = String.valueOf(Integer.valueOf(Math.floor(Math.random() * 100)));
+            }
+            if (Integer.valueOf(currContact.DummyJSON_Id__c) <= 100) {
+                DummyJSONUserFromId(currContact.DummyJSON_Id__c);
```

Comment:
Since you're still in the before insert context here, you don't want to make a callout. Callouts need to happen in after contexts to ensure that the records we're working with have been saved and no rollbacks/validation errors occurred.